### PR TITLE
Stable Operators Deployment for OCP Upgrade 

### PR DESCRIPTION
**Pull request by ebattat**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] dependencies

## Description

1. Add the ability to install stable operators: LSO, ODF, and CNV for OCP upgrades.
2. Update LSO/ODF to install from the stable channel and use installPlanApproval: Automatic.
3. Add a wait_for_csv method with number validation and add a custom error: CSVNotCreateTimeout.
4. Implement a double-check for operator installation validation, which is mandatory for upgrades between operator internal releases.
5. If the operator installation fails, trigger the PerfCI_OpenShift_Deployment Jenkinsfile.

### CODE REVIEW

File: benchmark_runner/common/oc/oc.py [hunk 1]
```diff
@@ -654,6 +654,27 @@ def wait_for_initialized(self, label: str, workload: str = '', status: str = 'In
...
+        :param csv_num: number of csvs, default 1
+        :param namespace:
+        :param timeout:
+        :return: csv names if getting csv or raise CSVError
+        """
+        current_wait_time = 0
+        while timeout <= 0 or current_wait_time <= timeout:
+            csv_names = self.run(f"{self.__cli} get csv -n {namespace} -o jsonpath={{$.items[*].metadata.name}}")
+            if csv_names and len(csv_names.split()) == csv_num:
+                return csv_names
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:672]: Need to handle the case of getting more CSVs than you expect.

Also, do you care what phase they're in at this point?

File: benchmark_runner/common/ocp_resources/create_odf.py [hunk 2]
```diff
@@ -52,14 +54,9 @@ def create_odf(self):
                                                       verify_cmd=r"""oc get pv -o jsonpath="{range .items[*]}{.metadata.name}{'\n'}{end}" | grep local | wc -l""",
                                                       count_openshift_storage=True)
                 if '07_subscription.yaml' in resource:
-                    # wait till get the patch
-                    self.wait_for_ocp_resource_create(resource='odf',
-                                                      verify_cmd="oc get InstallPlan -n openshift-storage -ojsonpath={.items[0].metadata.name}",
-                                                      status="install-")
-                    self.apply_patch(namespace='openshift-storage', resource='odf')
-                elif '08_storage_cluster.yaml' in resource:
                     # Must be run after installing the storage cluster because CSVs sometimes fail
```
Comment (RobertKrawitz) [inline benchmark_runner/common/ocp_resources/create_odf.py:57]: Looks like this comment has effectively moved from 08_storage_cluster.yaml to 07_subscription.yaml.

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]

File: benchmark_runner/common/oc/oc.py (hunk 1 above)
Comment (ebattat) [reply benchmark_runner/common/oc/oc.py:672]: csv_num is a dynamically number according to relevant operator, the default is 1

File: benchmark_runner/common/ocp_resources/create_odf.py (hunk 2 above)
Comment (ebattat) [reply benchmark_runner/common/ocp_resources/create_odf.py]: Yes, in the past we used installPlanApproval: Manual and I have changed it to installPlanApproval: Automatic, that cause that issue, remove this comment.

File: benchmark_runner/common/oc/oc.py (hunk 1 above)
Comment (RobertKrawitz) [reply benchmark_runner/common/oc/oc.py:672]: What happens if `oc get csv...` returns _more_ CSVs than what you expect, but includes everything you do want?

Comment (ebattat) [reply benchmark_runner/common/oc/oc.py:672]: I can change it to:
`if csv_names and len(csv_names.split()) >= csv_num:`
WDYT ?

Comment (RobertKrawitz) [reply benchmark_runner/common/oc/oc.py:672]: I think ideally you'd check to make sure that you have all of the specific CSVs that you want, but short of that, I think >= would be a better test.

Comment (ebattat) [reply benchmark_runner/common/oc/oc.py]: Done

**Review by RobertKrawitz — APPROVED** [review]

**Pull request merged**